### DISCUSSION TITLE
Kong using old upstream IPs

### DISCUSSION THREAD

**Discussion by ksgnextuple** [Kong/kong#11676] [Help]
Hi All

We face an issue where post new deployments (Basically when new pods come in) Kong continues to try to send the requests to old upstream IPs. During the same deployment time window we also see an error on kong which states - 

**2023/09/13 06:07:04 [error] 1174#0: 37405478 [lua] handler.lua:628: get_updated_plugins_iterator(): could not rebuild plugins iterator: [postgres] FATAL: sorry, too many clients already (stale plugins iterator will be used), context: ssl_certificate_by_lua,**

On analysing further we also see the target table in postgres is updated with new IPs and does not contain the stale IPs. We are guessing the cache in Kong is not refreshed and hence the issue.

As a temp solution we have restarted the kong pods.

What could be a permanent fix?
We are thinking of incresing the postgres max connections config. Is there anything we could do from kong side?

**Comment by StarlightIbuki** [comment]
This looks like a bug. I'm creating an issue to track this.

**Reply by StarlightIbuki** [reply]
https://github.com/Kong/kong/issues/12015

**Reply by ksgnextuple** [reply]
We increased the postgres max connections configuration and have not faced the issue again so far.

**Reply by StarlightIbuki** [reply]
@ksgnextuple Glad to know we have a resolution for the issue. Could you share the number of connections being used?
